### PR TITLE
Update sessions.py

### PR DESCRIPTION
**Pull request by alanyee**
Use `.format()` over `%` given PEP 3101

### CODE REVIEW

**Comment by kobayashi** [comment]
This is great for python3. @kenneth-reitz, I just want to make sure [requests](https://github.com/kennethreitz/requests) is continuing to spport python2.7. right?

**Comment by alanyee** [comment]
`.format()` should work for Python 2.7.

File: requests/sessions.py [hunk 1]
```diff
@@ -728,7 +728,7 @@ def get_adapter(self, url):
                 return adapter
 
         # Nothing matches :-/
-        raise InvalidSchema("No connection adapters were found for '%s'" % url)
+        raise InvalidSchema("No connection adapters were found for {!r}".format(url))
```
Comment (bisguzar) [inline requests/sessions.py:731]: What **!r** stands for? Python 2.7 supporting .format as well as you tell. Its ok but I couldn't get first one.

Comment (alanyee) [reply requests/sessions.py:731]: `{!r}` in `.format()` is the equivalent to `%r` in  `%`.

Comment (bisguzar) [reply requests/sessions.py:731]: The original line has 's' flag, why you used 'r'. I don't have enough knowledge about format flags

Comment (alanyee) [reply requests/sessions.py:731]: `url` should always be a string. So far `url = 'https://example.com/`, `"No connection adapters were found for '{!s}'".format(url)` is the same as `"No connection adapters were found for {!r}".format(url)`, as both output `"No connection adapters were found for 'https://example.com/'`,  for the raw form of a string includes its quotes. `{!r}` or `repr()` tends to be faster than `{!s}` or `str()`.

**Comment by kennethreitz** [comment]
interesting fact: % is much faster (at least in my tests, years ago) than `.format`. I don't think we need to be concerned about that here though ;)

**Pull request merged**